mastodon/src/main/java/org/joinmastodon/android/ui/text/ClickableLinksDelegate.java

```diff
@@ -63,6 +71,7 @@ public boolean onTouch(MotionEvent event) {
 							}
 							hlPath=new Path();
 							selectedSpan=span;
+							longClickHandler.postDelayed(copyTextToClipboard, ViewConfiguration.getLongPressTimeout());
```

Comment:
IMO since you already have the TextView here, it's better to call `postDelayed()` on it instead of using a separate handler just for the delay.

But then I'd use a [GestureDetector](https://developer.android.com/reference/android/view/GestureDetector?hl=en) because on some devices, like my Pixel 4a, it allows invoking the long-tap action faster by pressing harder.

Reply:
I thought about the GestureDetector, but I didn't want to rewrite the whole class... I will see, what I can do.

Reply:
A clean implementation of a guesture detector requires [`
GestureDetectorCompat`](https://developer.android.com/reference/androidx/core/view/GestureDetectorCompat) from `androidx.core:core`. This package, however, conflicts with `me.grishka.litex:collection`. An implementation without GestureDetectorCompat doesn't make too much sense, because this is what provides the `onTouchEvent` function that automatically detects the event-type.

Reply:
No, you don't need androidx. We don't use support libraries in this project and you shouldn't modify `build.gradle`. You should use the system APIs directly.

>This compatibility implementation of the framework's GestureDetector guarantees the newer focal point scrolling behavior from Jellybean MR1 on all platform versions.

And the oldest Android version the app supports is Marshmallow (6.0).

Reply:
Okay but without the GuestureDetectorCompat, what I have is more or less the closest I can get. The long press detection in the GuestureDetectorCompat also only checks timestamps against the `ViewConfiguration.getLongPressTimeout()`. I have no idea, how Google maps the "firm press" on pixel phones.

Reply:
I believe [this is the relevant place in AOSP](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/GestureDetector.java;l=728)

Reply:
I moved the postDelayed to the view and removed the Handler object.

Reply:
Okay it might be a little easier with a GestureDetector. But with it, I will have to manually enforce the ACTION_CANCEL event to reset the highlighting rectangle.

Reply:
I just pushed an implementation with a gesture listener.